lib/rules/validate-order-in-object-keys.js

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Validates the order in object keys.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values:
+ *  - `true` (alias to `asc`)
+ *  - `"asc"`: requires sorting in ascending order
+ *  - `"asc-insensitive"`: requires sorting in ascending order (case-insensitive)
+ *  - `"asc-natural"`: requires sorting in ascending natural order
+ *  - `"desc"`: requires sorting in descending order
+ *  - `"desc-insensitive"`: requires sorting in descending order (case-insensitive)
+ *  - `"desc-natural"`: requires sorting in descending natural order
+ *
+ * #### Example
+ *
+ * ```js
+ * "validateOrderInObjectKeys": "asc"
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var x = {
+ *  x: 'foo',
+ *  y: 'bar'
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = {
+ *  y: 'foo',
+ *  x: 'bar'
+ * }
+ * ```
+ */
+
+require('natural-compare-lite');
+
+var assert = require('assert');
+
+/**
+ * Sort in ascending order.
+ */
+function asc(a, b) {
+    return String(a) < String(b) ? -1 : 1;
+}
+
+/**
+ * Sort in ascending order (case-insensitive).
+ */
+function ascInsensitive(a, b) {
+    var lowercaseA = String(a).toLowerCase();
+    var lowercaseB = String(b).toLowerCase();
+
+    if (lowercaseA < lowercaseB) {
+        return -1;
+    }
+
+    if (lowercaseA > lowercaseB) {
+        return 1;
+    }
+
+    return asc(a, b);
+}
+
+/**
+ * Natural sort in ascending order.
+ */
+function ascNatural(a, b) {
+    return String.naturalCompare(a, b);
```

Comment:
@markelog unless I'm missing something, it looks like the `natural-sort` npm package is frontend only (https://github.com/studio-b12/natural-sort/blob/master/dist/natural-sort.js). The sort function is not exported.

If you meant https://github.com/javve/natural-sort, unfortunately it's not available on npm (only component). I've given https://www.npmjs.com/package/natural-compare a go and it seems to be exactly like the original module but without the native prototype intrusion. It doesn't look like it's actively maintained, but time will tell. Rebased with master.